xcube_smos/nckcindex/nckcindex.py

```diff
@@ -134,44 +135,36 @@ def create(
             source_storage_options=source_storage_options,
         )
 
-        if replace_existing and index_fs.isdir(index_path):
-            index_fs.rm(index_path, recursive=True)
-        index_fs.mkdirs(index_path)
-        with index_fs.open(cls._index_config_path(index_path), "w") as f:
-            json.dump(index_config, f, indent=2)
-        return cls.open(index_path,
-                        index_storage_options=index_storage_options)
+        index_store = IndexStore.new(index_path, mode="x", replace=replace)
+        index_store.write(INDEX_CONFIG_FILENAME,
+                          json.dumps(index_config, indent=2))
+        index_store.close()
+
+        return cls.open(index_path, mode="a")
 
     @classmethod
-    def open(
-        cls,
-        index_path: Union[str, Path],
-        index_protocol: Optional[str] = None,
-        index_storage_options: Optional[Dict[str, Any]] = None
-    ) -> "NcKcIndex":
+    def open(cls, index_path: Union[str, Path], mode: str = "r") \
+            -> "NcKcIndex":
         """Open the given index at *index_path*.
 
         :param index_path: The index path or URL.
-        :param index_protocol: The index protocol.
-            If not provided, it will be derived from *index_path*.
-        :param index_storage_options: Optional storage options for accessing
-            the filesystem of *index_path*.
-            See Python fsspec package spec for the used index protocol.
+        :param mode: Open mode, must be either "w" or "r".
+            Defaults to "r".
```

Comment:
Contradicts the implementation below, which allows "a" in addition to "w" and "r".